nexoan/crud-api/db/queryExecutor/Interfaces/executor.go

```diff
@@ -0,0 +1,454 @@
+//		Operation: "get_metadata",
+//		Filters: map[string]interface{}{
+//			"entityId": "123",
+//		},
+//	})
+type Executor interface {
+	// Execute processes a query based on the provided QueryParameter.
+	// It routes the query to the appropriate database implementation.
+	//
+	// Parameters:
+	//   - ctx: Context for the operation
+	//   - plan: QueryParameter containing operation details
+	//
+	// Returns:
+	//   - interface{}: Result of the operation, type depends on the operation
+	//   - error: Any error that occurred during execution
+	//
+	// Example:
+	//   result, err := executor.Execute(ctx, &QueryParameter{
+	//       Operation: "get_metadata",
+	//       Filters: map[string]interface{}{
+	//           "entityId": "123",
+	//       },
+	//   })
+	Execute(ctx context.Context, plan *QueryParameter) (interface{}, error)
+}
+
+// QueryParameter defines the structure for query execution parameters.
+// It provides a standardized way to specify operation details across different database types.
+//
+// Example:
+//
+//	param := &QueryParameter{
+//		Operation: "get_metadata",
+//		Filters: map[string]interface{}{
+//			"entityId": "123",
+//			"kind": "Person",
+//		TimeRange: struct{
+//			StartTime: "2024-01-01T00:00:00Z",
+//			EndTime: "2024-12-31T23:59:59Z",
+//		},
+//	}
+type QueryParameter struct {
+	// Operation specifies the type of operation to perform
+	// Valid values include:
+	// - "get_metadata": Retrieve entity metadata
+	// - "create_entity": Create a new entity
+	// - "get_relationships": Get entity relationships
+	Operation string
+
+	// Filters specify criteria for the operation
+	// Common filter keys:
+	// - "entityId": ID of the entity
+	// - "kind": Entity kind/type
+	// - "name": Entity name
+	Filters map[string]interface{}
+
+	// TimeRange specifies temporal constraints for the operation
+	TimeRange struct {
+		// StartTime is the ISO8601 timestamp for the start of the time range
+		StartTime string
+
+		// EndTime is the ISO8601 timestamp for the end of the time range
+		EndTime string
+
+		// ActiveAt is the ISO8601 timestamp to check if entity/relationship was active
+		ActiveAt string
+	}
+}
+
+// DocumentExecutor interface defines operations for document-based databases (e.g., MongoDB).
+// It handles metadata operations for entities.
+//
+// Example Usage:
+//
+//	executor := GetDocumentExecutor()
+//	metadata, err := executor.GetMetadata(ctx, &QueryParameter{
+//		Filters: map[string]interface{}{
+//			"entityId": "123",
+//		},
+//	})
+type DocumentExecutor interface {
+	Executor
+
+	// HandleMetadata manages entity metadata operations.
+	//
+	// Parameters:
+	//   - ctx: Context for the operation
+	//   - plan: QueryParameter containing metadata operation details
+	//
+	// Returns:
+	//   - error: Any error that occurred during the operation
+	//
+	// Example:
+	//   err := executor.HandleMetadata(ctx, &QueryParameter{
+	//       Operation: "update_metadata",
+	//       Filters: map[string]interface{}{
```

Comment:
```suggestion
	//       Parameters: map[string]interface{}{
```

Please change this everywhere else we have used `Filters`